### ISSUE TITLE
Possibly wrong type attribute for numeric values

### ISSUE THREAD

**Issue by rabanti-github** [rabanti-github/NanoXLSX#9]
Numeric values are usually described as follows:
```
    <row r="1" spans="1:1" x14ac:dyDescent="0.25">
      <c r="A1">
        <v>0</v>
      </c>
    </row>
```
However, NanoXLSX and PicoXLSX seems so generate this:
```
<row r="7">
      <c t="s" r="A7">
        <v>7</v>
      </c>
      <c t="n" r="B7">
        <v>0</v>
      </c>
...
    </row>
```
Issues are:
- The "t" attribute seems to be wrong or outdated
- Excel seems to generate one row tag per cell (maybe not optimized)

The datatypes of worksheets must be investigated and aligned to the current OOXML specs.

**Comment by rabanti-github** [comment]
Specs regarding cell types are located in the ECMA resource: [ECMA-376 5th edition Part 1](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-376,%20Fifth%20Edition,%20Part%201%20-%20Fundamentals%20And%20Markup%20Language%20Reference.zip), Page 2451.

Some considerations:
- 'd' (Date) is specified as "ISO 8601" value. However, have never seen such a format in actual worksheets (assumption: date defined as string). The usage of 'd' in the writer may be wrong
- 'n' (Number) is specified as common number. However, Excel seems simply to omit the attribute in case of numbers (legacy definition or quirk?)
- 's' (Shared String) accepts either numbers as reference to the shared strings, or text. The later one was an old implementation of PicoXLSX. Not sure whether inline text with attribute value 's' instead of 'inlineStr' is an undocumented quirk or some type of fallback. However, not really relevant since shared strings are used always.

The PR will fix an obvious casting error.

A future improvement will be to revise the type handling in the Writer for the attribute values:
- 'd' (probably to remove or convert the value to ISO string instead of OADate)
- 'n' (probably completely to remove in the writer, and to keep in the reader for compatibility reason)
- 's' and 'str' (check the proper use of the attribute values)

**Comment by rabanti-github** [comment]
Case closed with release ([v1.4.1](https://github.com/rabanti-github/NanoXLSX/releases/tag/1.4.1).
The NuGet package should be available in some minutes.
The further improvements, as mentioned above, will be implemented either in a new minor or major version (not scheduled)
@wuzzeb Thank you for the contribution.
